### ISSUE TITLE
Show all release notes in the upgrade window

### ISSUE THREAD

**Issue by lwouis** [lwouis/alt-tab-macos#225]
[There is a bug currently in Sparkle](https://github.com/sparkle-project/Sparkle/issues/985#issuecomment-611081826) where it shows only releases notes for the latest release, instead of the release notes of all releases between current and latest.

A workaround would be to display the full changelog on all releases. I think it's better than the current state. See the discussion in the link above for more details

**Comment by lwouis** [comment]
There are currently 680 lines in the changelog. I'm worried that the `appcast.xml` file could get too big too fast if we put the full changelog on every release. The file can get into the multi-megabytes quickly, and then download time will be long, and displaying the text may be reducing UI performance in the textview.

**Comment by DaVince** [comment]
Thanks for this app, it's very useful and nice :)

I have no experience of knowledge of what Sparkle handles exactly, but I can think of the following options to limit the appcast.xml size as follows:

- Limit the amount of entries to 10 max (if possible).
- Show the full changelog text for the very last version only.
- Limit older entries to a 5 line summary at most, with a link at the end pointing to that version's Github release page in case someone wants the full changelog.

**Comment by lwouis** [comment]
@DaVince thanks for the suggestions!

All 3 of them have the downside that we have to implement that, and that involves either risky regexps, or involving a markdown or xml parser then writing transformation code.

The solution I have in mind for now, that would be very easy to implement is to use Sparkle `appcast.xml` tag called `releaseNotesLink`. Using this tag instead of inlining the changelog delta as is done today, Sparkle will display the full release note on every upgrade, but the appcast only list 1 line every release, like:

```xml
<sparkle:releaseNotesLink>https://lwouis.github.io/alt-tab-macos/CHANGELOG</sparkle:releaseNotesLink>
```

To have a clean HTML page though, I can't just link to the CHANGELOG.md from github, otherwise there is github framing around. I'm waiting on #211 to be completed, and then I will link to the CHANGELOG from there, which will be fullscreen and nice. I already experimented with that and the results are very nice.

**Comment by lwouis** [comment]
I've released the new website, and also a [new page](https://alt-tab-macos.netlify.app/changelog-bare) which is displayed by Sparkle on update. I find this new UX very nice and clear. The only missing piece would be localization, but that's a huge undertaking. We could imagine providing google translate links though. However I'm not sure that would be an actual improvement, if the translation is worse than showing english for some users who understand a bit of english.

**Comment by lwouis** [comment]
I thought it was done, but there is a very complex issue in Sparkle which makes it so the translation of release notes is not working currently. I opened a [PR to Sparkle](https://github.com/sparkle-project/Sparkle/pull/1586). Hopefully they merge it and release a new version. I tried to fork my own but I don't understand how they package Sparkle. It's different from a simple pod like Shortcut Recorder which I forked in the past.
